contracts/modules/ProtocolSettings.sol

```diff
@@ -67,6 +67,48 @@ contract ProtocolSettings is State, ProtocolTokenUser, ProtocolSettingsEvents {
 		_setTarget(this.setProtocolTokenAddress.selector, target);
 		_setTarget(this.setRolloverBaseReward.selector, target);
 		_setTarget(this.setRebatePercent.selector, target);
+		_setTarget(this.setSovrynProtocolAddress.selector, target);
+		_setTarget(this.setSOVTokenAddress.selector, target);
+		_setTarget(this.setLockedSOVAddress.selector, target);
+		_setTarget(this.setMinReferralsToPayoutAffiliates.selector, target);
+	}
+
+	/**
+	 * setting wrong address will break inter module functions calling
+	 * should be set once
+	 */
+	function setSovrynProtocolAddress(address newProtocolAddress) external onlyOwner {
+		address oldProtocolAddress = protocolAddress;
+		protocolAddress = newProtocolAddress;
+
+		emit SetProtocolAddress(msg.sender, oldProtocolAddress, newProtocolAddress);
+	}
+
+	function setSOVTokenAddress(address newSovTokenAddress) external onlyOwner {
+		require(Address.isContract(newSovTokenAddress), "newSovTokenAddress not a contract");
+
+		address oldTokenAddress = sovTokenAddress;
+		sovTokenAddress = newSovTokenAddress;
+
+		emit SetSOVTokenAddress(msg.sender, oldTokenAddress, newSovTokenAddress);
+	}
+
+	function setLockedSOVAddress(address newLockedSOVAddress) external onlyOwner {
+		require(Address.isContract(newLockedSOVAddress), "newLockSOVAddress not a contract");
+
+		address oldLockedSOVAddress = lockedSOVAddress;
+		lockedSOVAddress = newLockedSOVAddress;
+
+		emit SetLockedSOVAddress(msg.sender, oldLockedSOVAddress, newLockedSOVAddress);
+	}
+
+	function setMinReferralsToPayoutAffiliates(uint256 newMinReferrals) external onlyOwner {
+		require(newMinReferrals > 0, "Minimum referrals must be greater than 0");
```

Comment:
what if we will need to cancel this limitation?
```suggestion
		require(newMinReferrals > 0, "Minimum referrals must be greater than 0");
```
```suggestion
```

Reply:
If cancel the limitation is needed, then i will remove the require